src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -49,9 +51,12 @@ public FindCommand parse(String args) throws ParseException {
                 PREFIX_EMAIL, PREFIX_GENDER, PREFIX_TAG, PREFIX_LOCATION, PREFIX_TYPE, PREFIX_GITHUBUSERNAME,
                 PREFIX_RATING, PREFIX_SPECIALISATION, PREFIX_YEAR, PREFIX_OFFICEHOUR);
 
-        if (!(areAllArgsValid(PREFIX_NAME, PREFIX_MODULE_CODE, PREFIX_PHONE, PREFIX_EMAIL,
+        if (!(areAllArgsNonEmpty(PREFIX_NAME, PREFIX_MODULE_CODE, PREFIX_PHONE, PREFIX_EMAIL,
                 PREFIX_GENDER, PREFIX_TAG, PREFIX_LOCATION, PREFIX_TYPE, PREFIX_GITHUBUSERNAME,
                 PREFIX_RATING, PREFIX_SPECIALISATION, PREFIX_YEAR, PREFIX_OFFICEHOUR))) {
+            if (hasEmptyFields) {
```

Comment:
Why do you need this additional if statement. Is it redundant?

Reply:
It's not redundant, but the method `areAllArgsNonEmpty` now has a misleading name and its implementation looks strange. I will change it.

`areAllArgsNonEmpty` is supposed to check if at least one field has been specified, and then after that, `hasEmptyFields` is supposed to check if any of the fields provided are empty.